yt/frontends/ramses/data_structures.py

```diff
@@ -857,9 +867,59 @@ def read_namelist(self):
 
             self.parameters["namelist"] = nml
 
+    @staticmethod
+    def _sanitize_filename(path):
```

Comment:
I think this function is complex enough to warrant a class definition. It would also allow its members (functions) be be unit-tested